common-tasks/block-devices.md

```diff
@@ -0,0 +1,168 @@
+
+If several different block-devices are attached to a single VM, the last letter of the device node name is advanced through the alphabet, so after `xvdi` the next device will be named `xvdj`, the next `xvdk`, and so on.
+
+To specify this device node name, you can pass `--option frontend-dev=[custom-node-name]` to `qvm-block attach`.
+
+#Command Line Tool Guide
+The command-line tool you may use to mount whole USB drives or their partitions is `qvm-block`, a shortcut for `qvm-device block`.
+
+`qvm-block` only sees device-nodes and may not use names you expect! So make sure to have the drive available in the sourceVM before listing available block devices (step 1.) to find out it's its ID.
+
+In case of a USB-drive, make sure it's attached to your computer. If you don't see anything that looks like your drive, run `sudo udevadm trigger --action=change` in your USB-qube (typically `sys-usb`)
+
+ 1. In a dom0 console (running as a normal user), list all available block devices:
+    
+        qvm-block
+    
+    This will list all available block devices in your system across all VMs, no matter whether hosted by a USB controller or `losetup`.
+    The name of the qube hosting the block device is displayed before the colon in the device ID.
+    The string after the colon is the ID of the device used within the qube, like so:
+
+        sourceVM:sdb     Cruzer () 4GiB
+        sourceVM:sdb1    Disk () 2GiB
+
+ 2. Assuming your block device is attached to `sys-usb` and its device node is `sdb`, we attach the device to a qube with the name `work` like so:
+    
+        qvm-block attach work sys-usb:sdb
+    
+    This will attach the device to the qube as `/dev/xvdi` if that name is not already taken by another attached device, or `/dev/xvdj`, etc.
+    
+    You may also mount one partition at a time by using the same command with the partition number, e.g. `sdb1`.
+
+ 3. The block device is now attached to the qube.
+    If using a default qube, you may open the Nautilus file manager in the qube, and your drive should be visible in the **Devices** panel on the left.
+    If you've attached a single partition (e.g. `sdb2` instead of `sdb` in our example), you may need to manually mount before it becomes visible:
+    
+        cd ~
+        mkdir mnt
+        sudo mount /dev/xvdi mnt
+    
+
+ 4. When you finish using the block device, click the eject button or right-click and select **Unmount**.
+
+    If you've manually mounted a single partition in the above step, use:
+
+        sudo umount mnt
+
+ 5. In a dom0 console, detach the device
+
+        qvm-block detach work sys-usb:sdb
+
+ 6.  You may now remove the device or attach it to another qube.
+<!--TODO: what happens if USB-device removed before detaching? How about attaching a blockdevice to two qubes?-->
```

Comment:
That part of the documentation is specifically for 3.2, so I ignored it for 4.0. Is there a version for 4.0 or is it equally valid, despite the title?

Reply:
Since [qubes-utils 4.0.6](https://github.com/marmarek/qubes-linux-utils/commit/764b0f3f0778c7675ded7aa67aa954917045f20a) much simpler recovery is possible:
```
virsh detach-disk work xvdi
```
(adjust `work` and `xvdi`)
Unless device is originally from dom0, then the R3.2 procedure needs to be used.